src/DataModel/IData.ts

```diff
@@ -0,0 +1,6 @@
+import {IMetadata} from "./IMetadata";
+
+export interface IData extends Object
```

Comment:
I agree that this separation might not come easily. I agree that library doesn't know what higher level libraries, apps and UIs want to do with the data. I still think that client should know what is a hypermedia control - it's supposed to consume those.
Anyway - from API point of view - client should expose somehow both the payload (separated or not) and hypermedia controls it discovers within that payload.
@tpluscode example is a good one in that matter.